pkg/volume/photon_pd/photon_util.go

```diff
@@ -0,0 +1,144 @@
+
+type PhotonDiskUtil struct{}
+
+func removeFromScsiSubsystem(volName string) {
+	// TODO: if using pvscsi controller, this won't be needed
+	deviceName := volNameToDeviceName[volName]
+	fileName := "/sys/block/" + deviceName + "/device/delete"
+	data := []byte("1")
+	ioutil.WriteFile(fileName, data, 0666)
+}
+
+func scsiHostScan() {
+	// TODO: if using pvscsi controller, this won't be needed
+	scsi_path := "/sys/class/scsi_host/"
+	if dirs, err := ioutil.ReadDir(scsi_path); err == nil {
+		for _, f := range dirs {
+			name := scsi_path + f.Name() + "/scan"
+			data := []byte("- - -")
+			ioutil.WriteFile(name, data, 0666)
+			glog.Errorf("scsiHostScan scan for %s", name)
+		}
+	}
+}
+
+func verifyDevicePath(path string) (string, error) {
+	if pathExists, err := volumeutil.PathExists(path); err != nil {
+		return "", fmt.Errorf("Error checking if path exists: %v", err)
+	} else if pathExists {
+		return path, nil
+	}
+
+	glog.V(4).Infof("verifyDevicePath: path not exists yet")
+	return "", nil
+}
+
+// CreateVolume creates a PhotonController persistent disk.
+func (util *PhotonDiskUtil) CreateVolume(p *photonPersistentDiskProvisioner) (pdID string, capacityGB int, err error) {
+	cloud, err := getCloudProvider(p.plugin.host.GetCloudProvider())
+	if err != nil {
+		glog.Errorf("Photon Controller utility: CreateVolume failed to get cloud provider")
+		return "", 0, err
+	}
+
+	capacity := p.options.PVC.Spec.Resources.Requests[api.ResourceName(api.ResourceStorage)]
+	volSizeBytes := capacity.Value()
+	// PhotonController works with GB, convert to GB with rounding up
+	volSizeGB := int(volume.RoundUpSize(volSizeBytes, 1024*1024*1024))
+	name := volume.GenerateVolumeName(p.options.ClusterName, p.options.PVName, 255)
+	volumeOptions := &photon.VolumeOptions{
+		CapacityGB: volSizeGB,
+		Tags:       *p.options.CloudTags,
+		Name:       name,
+	}
+
+	for parameter, value := range p.options.Parameters {
+		switch strings.ToLower(parameter) {
+		case "flavor":
+			volumeOptions.Flavor = value
+		default:
+			return "", 0, fmt.Errorf("invalid option %q for volume plugin %s", parameter, p.plugin.GetPluginName())
+		}
+	}
+
+	pdID, err = cloud.CreateDisk(volumeOptions)
+	if err != nil {
+		glog.Errorf("Photon Controller utility: failed to CreateDisk, error [%v]", err)
+		return "", 0, fmt.Errorf("Photon Controller utility: failed to CreateDisk, error [%v]", err)
+	}
+
+	glog.V(4).Infof("Successfully created Photon Controller persistent disk %s", name)
+	return pdID, volSizeGB, nil
+}
+
+// DeleteVolume deletes a vSphere volume.
+func (util *PhotonDiskUtil) DeleteVolume(pd *photonPersistentDiskDeleter) error {
+	cloud, err := getCloudProvider(pd.plugin.host.GetCloudProvider())
+	if err != nil {
+		glog.Errorf("Photon Controller utility: DeleteVolume failed to get cloud provider")
+		return err
+	}
+
+	if err = cloud.DeleteDisk(pd.pdID); err != nil {
+		glog.Errorf("Photon Controller utility: failed to DeleteDisk for pdID %s, error [%v]", pd.pdID, err)
```

Comment:
Following will look better,
formated_err := fmt.Errorf("Photon Controller utility: failed to DeleteDisk for pdID %s, error [%v]", pd.pdID, err)
glog.Error(formated_err)
return err